sites/automationworld/redirect-handler.js

```diff
@@ -0,0 +1,19 @@
+const digital = async (from) => {
+  const pattern = /\/sites\/default\/files\/digital_edition\/(.*)/;
+  const matches = pattern.exec(from);
+  if (matches && matches[1]) {
+    return { to: `https://digitaleditions.automationworld.com/${matches[1]}` };
+  }
+  return null;
+};
+
+const s3 = async (from) => {
+  const pattern = /\/sites\/default\/files\/(.*)/;
+  const matches = pattern.exec(from);
+  if (matches && matches[1]) {
+    return { to: `https://s3.us-east-2.amazonaws.com/pmg-production/Migrated+-+DO+NOT+USE/AW/${matches[1]}` };
+  }
+  return null;
+};
+
+module.exports = async ({ from }) => Promise.all([digital(from), s3(from)]).then(l => l[0]);
```

Comment:
Doing a `Promise.all` in this way will _always_ return the first value. As such, you're `s3` redirect will never be returned.

That being said, Promises aren't needed for these functions, nor is `Promise.all`